src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,222 @@
+                    return "Wrong input format";
+                }
+            case "delete":
+                if (infos.length == 2) {
+                    String index = infos[1];
+                    return deleteTask(index, taskList);
+                } else {
+                    return "Wrong input format";
+                }
+            default:
+                return addList(input, taskList);
+        }
+    }
+
+
+    /**
+     * Returns error messages or a string showing the added task.
+     * The input string must follow input format, otherwise error messages will be return.
+     * The new task will be created and added to lst.
+     * @param input input message
+     * @param taskList the taskList containing all tasks
+     * @return a string showing the added task and number of tasks
+     */
+     static String addList(String input, TaskList taskList) {
+         ArrayList<Task> lst = taskList.getTasks();
+         ArrayList<String> validType = new ArrayList<>(
+                 Arrays.asList("deadline", "event", "todo"));
+         String type = input.split(" ", 2)[0];
+         String content;
+         if (!validType.contains(type)) {
+             return "I'm sorry, but I don't know what that means :-(";
+         }
+         try {
+             content = input.split(" ", 2)[1];
+         } catch (IndexOutOfBoundsException e) {
+             return "The description of a todo cannot be empty.";
+         }
+
+         if (type.equals("todo")) {
+             lst.add(new Todo(content));
+         } else if (type.equals("deadline")) {
+             String[] strings = content.split(" /by ");
+             if (strings.length != 2) {
+                 return "Please check the format of your deadline.";
+             }
+             lst.add(new Deadline(content.split(" /by ")[0], content.split( " /by ")[1]));
+         } else if (type.equals("event")) {
+             String[] strings = content.split(" /at ");
+             if (strings.length != 2) {
+                 return "Please check the format of your event.";
+             }
+             lst.add(new Event(content.split(" /at ")[0], content.split(" /at ")[1]));
+         }
+
+         String output = "    ____________________________________________________________\n"
+                 + "     Got it. I've added this task: \n"
+                 + "      " + lst.get(lst.size() - 1).toString() + "\n"
+                 + "     Now you have " + lst.size() +" tasks in the list. \n"
+                 + "    ____________________________________________________________\n";
+         return output;
+     }
+
+
+    /**
+     * Returns the message that shows which task is marked as completed.
+     * Mark the task indicated by index as completed.
+     * @param index the index of task to be marked
+     * @param taskList the taskList containing all tasks
+     * @return the messages
+     */
+    static String doTask(String index, TaskList taskList) {
+        int idx;
```

Comment:
```suggestion
        int indexToMarkComplete;
```
Would it be better to use standard words in your variable naming here? Also might be good to specify a more descriptive variable name. See '[Use standard words](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/useStandardWords/)' & '[Use name to explain](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/intermediate/useNameExplain/)' in the textbook.